iDiff/cmd/root.go

```diff
@@ -27,35 +27,44 @@ var RootCmd = &cobra.Command{
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
+
 		utils.SetDockerEngine(eng)
-		image1, err := utils.ImagePrepper{args[1]}.GetImage()
+
+		img1Arg := args[len(args)-2]
+		img2Arg := args[len(args)-1]
+		diffArgs := args[:len(args)-2]
+
+		image1, err := utils.ImagePrepper{img1Arg}.GetImage()
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
-		image2, err := utils.ImagePrepper{args[2]}.GetImage()
+		image2, err := utils.ImagePrepper{img2Arg}.GetImage()
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
-		differ, err := differs.GetDiffer(args[0])
+		diffTypes, err := differs.GetDiffers(diffArgs)
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
 
-		diff := differs.DiffRequest{image1, image2, differ}
-		if diff, err := diff.GetDiff(); err == nil {
-			if json {
-				err = diff.OutputJSON()
-				if err != nil {
-					glog.Error(err)
-				}
-			} else {
-				err = diff.OutputText()
-				if err != nil {
-					glog.Error(err)
+		req := differs.DiffRequest{image1, image2, diffTypes}
+		if diffs, err := req.GetDiff(); err == nil {
+			for _, diff := range diffs {
+				if json {
+					err = diff.OutputJSON()
```

Comment:
The json object for each diff are also indistinguishable if they are of the same type.  Maybe consider making another DiffResult type that can have it's own outputtext and outputjson so things can be properly labeled and less dependent on the individual differ outputs.